src/univers/version_constraint.py

```diff
@@ -210,8 +210,7 @@ def __contains__(self, version):
         >>> assert v24 in VersionConstraint(comparator="<=", version=v24)
         >>> assert v24 not in VersionConstraint(comparator="<", version=v24)
         """
-
-        if not isinstance(version, self.version.__class__):
+        if not isinstance(version, self.version.__class__) and self.comparator != "*":
```

Comment:
This does not seem to be the root cause.
Instead look at https://github.com/nexB/univers/blob/4dccdfb0041816fb840af9023ff3473490f02829/src/univers/version_range.py#L117
Here we are creating a "star" constraint passing a None class which is bug?

Reply:
Thanks for pointing out !